src/EditorFeatures/Core.Wpf/InlineRename/UI/Adornment/RenameFlyoutViewModel.cs

```diff
@@ -228,7 +231,9 @@ public bool Submit()
             }
 
             SmartRenameViewModel?.Commit(IdentifierText);
-            Session.Commit();
+
+            var token = _listener.BeginAsyncOperation(nameof(Submit));
+            _ = Session.CommitAsync(previewChanges: false).ReportNonFatalErrorAsync().CompletesAsyncOperation(token);
```

Comment:
note: we could still just have Commit be exposed synchronously on the session.  Just rename it to InitiateCommit and doc it that it will return immediately, after kicking off the tracked work to commit that it will then perform that will then run concurrently with the caller code.